addons/account/models/account_move_line.py

```diff
@@ -462,12 +462,13 @@ def _compute_currency_id(self):
     def _compute_name(self):
         for line in self.filtered(lambda l: l.move_id.inalterable_hash is False):
             if line.display_type == 'payment_term':
-                term_lines = line.move_id.line_ids.filtered(lambda l: l.display_type == 'payment_term') | line
+                term_lines = line.move_id.line_ids.filtered(lambda l: l.display_type == 'payment_term') | self.filtered(lambda l: l.display_type == 'payment_term' and l.move_id == line.move_id)
```

Comment:
In order to reduce time complexity (and I believe code complexity too), define
```python
terms_by_move = (self + line.move_id.line_ids).filtered(lambda l: l.display_type == 'payment_term').grouped('move_id')
```
Then you can just do
```python
term_lines = terms_by_move[line.move_id]
```